Solution/DesignAutomationInventorUtilities Tests/Helpers/NameValueMapHelperTests.cs

```diff
@@ -0,0 +1,225 @@
+            Assert.Throws<InvalidValueTypeException>(() => mapHelper.AsIntCollection("StringValue"));
+        }
+
+        [Fact]
+        public void TestHasValue()
+        {
+            NameValueMap nameValueMap = new NameValueMapStub();
+            nameValueMap.Value["StringValue"] = "TestString";
+            nameValueMap.Value["IntValue"] = "356";
+            nameValueMap.Value["DoubleValue"] = "114.3998";
+            nameValueMap.Value["BoolValue"] = "True";
+            nameValueMap.Value["EnumValue"] = "VALUE_FOUR";
+
+            NameValueMapHelper mapHelper = new NameValueMapHelper(nameValueMap);
+
+            Assert.True(mapHelper.HasKey("StringValue"));
+            Assert.True(mapHelper.HasKey("IntValue"));
+            Assert.True(mapHelper.HasKey("DoubleValue"));
+            Assert.True(mapHelper.HasKey("BoolValue"));
+            Assert.False(mapHelper.HasKey("NonExistantKey"));
+            Assert.True(mapHelper.HasKey("EnumValue"));
+        }
+
+        [Fact]
+        public void ConverterTest()
+        {
+            DataConverter dataConverter = new DataConverter();
+
+            // String
+            Assert.True(dataConverter.TryGetValueFromObjectAs("text", out string strResult));
+            Assert.Equal("text", strResult);
+
+            // Int + Numerical conversion fail
+            Assert.True(dataConverter.TryGetValueFromObjectAs("152", out int intResult));
+            Assert.Equal(152, intResult);
+            Assert.False(dataConverter.TryGetValueFromObjectAs("1662A", out intResult));
+
+            // Double
+            Assert.True(dataConverter.TryGetValueFromObjectAs("1444.73998", out double doubleResult));
+            Assert.Equal(1444.73998d, doubleResult);
+
+            // Bool
+            Assert.True(dataConverter.TryGetValueFromObjectAs("TRUE", out bool boolResult));
+            Assert.True(boolResult);
+            Assert.True(dataConverter.TryGetValueFromObjectAs("false", out boolResult));
+            Assert.False(boolResult);
+            Assert.True(dataConverter.TryGetValueFromObjectAs("True", out boolResult));
+            Assert.True(boolResult);
+            Assert.False(dataConverter.TryGetValueFromObjectAs("1", out boolResult));
+
+            // Enum
+            Assert.True(dataConverter.TryGetValueFromObjectAs("VALUE_FOUR", out TestEnum enumResult));
+            Assert.Equal(TestEnum.VALUE_FOUR, enumResult);
+            Assert.True(dataConverter.TryGetValueFromObjectAs("value_one", out enumResult));
+            Assert.Equal(TestEnum.VALUE_ONE, enumResult);
+        }
+
+        [Theory]
+        [ClassData(typeof(TestDataProvider))]
+        public void StringTest(NameValueMapHelper mapHelper, Dictionary<string, object> expectedResults)
+        {
+            string str1 = mapHelper.AsString("StringValue");
+            Assert.Equal(expectedResults["StringValue"], str1);
+        }
+
+        [Theory]
```

Comment:
Do we expect to run this test using more data variations? Maybe I understand it wrong, but now it seems DataProvider has only one item.